packages/tideline/plugins/blip/basics/logic/datamunger.js

```diff
@@ -31,8 +29,8 @@ function dataMunger(bgClasses, bgUnits = MGDL_UNITS) {
 
   return {
     getLatestPumpUploaded: function(patientData) {
-      const lastUploadDatum = getLatestPumpUpload(patientData.grouped.upload)
-      return _.get(lastUploadDatum, 'source', null)
+      const pumpSettings = patientData.grouped.pumpSettings[0]
```

Comment:
Are we always sure that `patientData.grouped.pumpSettings[0]` exists? Is there not a risk of an app crash if `patientData.grouped.pumpSettings` is null or empty?

Reply:
Given the implementation of the method `grouped()` (in `medical-data.service.ts`, line 126), as long as the `DatumType` "pumpSettings" exists, the value will at worst exist and be equal to an empty array. So worst case scenario, `patientData.grouped.pumpSettings[0]` will return `undefined`, but I don't think null pointer exception can happen here (please tell me if I'm mistaken!)

Reply:
Okay this works. I know that @ydrisR has made changes to this code too... To be kept in mind for his PR